### PR TITLE
fix(join) Fix case when an expression has multiple columns

### PR DESCRIPTION
**Pull request by evanh**
If an expression that needed to be moved to a branch involved two different
columns it would cause an error.

### CODE REVIEW

File: tests/test_snql_api.py [hunk 1]
```diff
@@ -510,3 +510,23 @@ def test_multi_table_join(self) -> None:
         )
 
         assert response.status_code == 200
+
+    def test_complex_table_join(self) -> None:
```
Comment (fpacifici) [inline tests/test_snql_api.py:514]: The branch cutter has a pretty extensive unit test base. Can we add the test there instead of adding an end to end test for this ?
https://github.com/getsentry/snuba/blob/master/tests/query/joins/test_branch_cutter.py#L19

Besides the performance gain in running the unit test over the integration test, that would be focused on covering this specific corner case, thus the unit test could not be made irrelevant by refactoring the way we manage joins. It would be also easier to pinpoint the issue in case of regression through a unit test than through an end to end test.

If you are short of time for this fix, I can take it over, no pb.

Comment (evanh) [reply tests/test_snql_api.py:514]: Added test.

**Review by fpacifici — APPROVED** [review]
thanks
